dedupe/labeler.py

```diff
@@ -69,24 +69,22 @@ def candidates(self) -> TrainingExamples:  # type: ignore[override]
     def candidates(self, new_candidates: TrainingExamples) -> None:
         self._candidates = new_candidates
 
-        self.distances = self.transform(self._candidates)
+        self.distances = self._distances(self._candidates)
 
         random_pair = random.choice(self._candidates)
         exact_match = (random_pair[0], random_pair[0])
-        self.fit_transform([exact_match, random_pair], [1, 0])
+        self.fit([exact_match, random_pair], [1, 0])
 
-    def transform(self, pairs: TrainingExamples) -> numpy.typing.NDArray[numpy.float_]:
+    def _distances(self, pairs: TrainingExamples) -> numpy.typing.NDArray[numpy.float_]:
         return self.data_model.distances(pairs)
 
-    def fit(self, X: numpy.typing.NDArray[numpy.float_], y: LabelsLike) -> None:
-
+    def _fit(self, X: numpy.typing.NDArray[numpy.float_], y: LabelsLike) -> None:
         self.y: numpy.typing.NDArray[numpy.int_] = numpy.array(y)
         self.X = X
```

Comment:
i know this was nor your design, but it's a bit weird to have the fit method be responsible for setting these instance attributes. `mark` would be a more natural place, imo

Reply:
I agree. still trying to figure out how to deal with the fact that the learners are stateful with their data. I almost want to split them up, so that one component is stateless like sklearn models and has a `fit()`, and that is owned by a learner that is stateful and remembers the old training data, and that has the `mark()`

Reply:
OK in this new version all the sublearners are stateless (in regards to saving training data) and don't have a `mark`, just a stateless fit(). Only DisagreementLEarner has a `mark` that stores traingin data.